content/typeclasses-syntax.md

```diff
@@ -0,0 +1,373 @@
+
+
+### 7. Abolish Abstract Givens
+
+Another simplification is possible. So far we have special syntax for abstract givens:
+```scala
+given x: T
+```
+The problem is that this syntax clashes with the quite common case where we want to establish a given without any nested definitions. For instance
+consider a given that constructs a type tag:
+```scala
+class Tag[T]
+```
+Then this works:
+```scala
+given Tag[String]()
+given Tag[String] with {}
+```
+But the following more natural syntax fails:
+```scala
+given Tag[String]
+```
+The last line gives a rather cryptic error:
+```
+1 |given Tag[String]
+  |                 ^
+  |                 anonymous given cannot be abstract
+```
+The problem is that the compiler thinks that the last given is intended to be abstract, and complains since abstract givens need to be named. This is another annoying dissonance. Nowhere else in Scala's syntax does adding a
+`()` argument to a class cause a drastic change in meaning. And it's also a violation of the principle that it should be possible to define all givens without providing names for them.
+
+Fortunately, abstract givens are no longer necessary since they are superseded by the new `deferred` scheme. So we can deprecate that syntax over time. Abstract givens are a highly specialized mechanism with a so far non-obvious syntax.
+We have seen that this syntax clashes with reasonable expectations of Scala programmers. My estimate is that maybe a dozen people world-wide have used abstract givens in anger so far.
+
+**Proposal** In the future, let the `= deferred` mechanism be the only way to deliver the functionality of abstract givens.
+
+This is less of a disruption than it might appear at first:
+
+ - `given T` was illegal before since abstract givens could not be anonymous.
+   It now means a concrete given of class `T` with no member definitions.
+ - `given x: T` is legacy syntax for an abstract given.
+ - `given T as x = deferred` is the analogous new syntax, which is more powerful since
+    it allows for automatic instantiation.
+ - `given T = deferred` is the anonymous version in the new syntax, which was not expressible before.
+
+**Benefits:**
+
+ - Simplification of the language since a feature is dropped
+ - Eliminate non-obvious and misleading syntax.
+
+The only downside is that deferred givens are restricted to be used in traits, whereas abstract givens are also allowed in abstract classes. But I would be surprised if actual code relied on that difference, and such code could in any case be easily rewritten to accommodate the restriction.
```

Comment:
No it would break the binary API. The thing is, I am not sure there is any use case in the wild where abstract givens are used in abstract classes. People usually reach for traits anyway.

Reply:
Then, as I mentioned before offline (but I can't find a written trace online), it is simply not acceptable to ever get rid of abstract givens. If there is a library that uses one today, it will never be able to upgrade its compiler version, no matter what source rewritings it attempts to do.

This is a big no-no. Removing them would not only break backward source compat (which we *can* do), it would break our promises of binary compatibility due to second-order effects.

We can *deprecate* them, with the usual options to silence the deprecation, but we can never remove them.